sdk/eventhub/event-hubs/samples/typescript/src/iothubConnectionString.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT Licence.
+
+/*
+  This sample demonstrates how to convert an Iot Hubs connection string to
+  an Event Hubs-compatible connection string that can be used with the EventHubConsumerClient
+  to process events from an Iot Hub.
+*/
+
+import * as crypto from "crypto";
+import { Buffer } from "buffer";
+import {
+  AmqpError,
+  Connection,
+  ReceiverEvents,
+  isAmqpError as rheaIsAmqpError,
+  parseConnectionString
+} from "rhea-promise";
+import { EventHubConsumerClient, earliestEventPosition } from "@azure/event-hubs";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+/**
+ * Type guard for AmqpError.
+ * @param err An unknown error.
+ */
+function isAmqpError(err: any): err is AmqpError {
+  return rheaIsAmqpError(err);
+}
+
+const consumerGroup = process.env["CONSUMER_GROUP_NAME"] || "";
+
+// This code is modified from https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens.
+function generateSasToken(
+  resourceUri: string,
+  signingKey: string,
+  policyName: string,
+  expiresInMins: number
+): string {
+  resourceUri = encodeURIComponent(resourceUri);
+
+  // Set expiration in seconds.
+  const expires = Math.ceil(Date.now() / 1000 + expiresInMins * 60);
+  const toSign = resourceUri + "\n" + expires;
+
+  // Use the crypto module to create the hmac.
+  const hmac = crypto.createHmac("sha256", Buffer.from(signingKey, "base64"));
+  hmac.update(toSign);
+  const base64UriEncoded = encodeURIComponent(hmac.digest("base64"));
+
+  // Construct authorization string.
+  return `SharedAccessSignature sr=${resourceUri}&sig=${base64UriEncoded}&se=${expires}&skn=${policyName}`;
+}
+
+/**
+ * Converts an IotHub Connection string into an Event Hubs-compatible connection string.
+ * @param connectionString An IotHub connection string in the format:
+ * `"HostName=<your-iot-hub>.azure-devices.net;SharedAccessKeyName=<KeyName>;SharedAccessKey=<Key>"`
+ * @returns An Event Hubs-compatible connection string in the format:
+ * `"Endpoint=sb://<hostname>;EntityPath=<your-iot-hub>;SharedAccessKeyName=<KeyName>;SharedAccessKey=<Key>"`
+ */
+async function convertIotHubToEventHubsConnectionString(connectionString: string): Promise<string> {
+  const { HostName, SharedAccessKeyName, SharedAccessKey } = parseConnectionString<{
```

Comment:
Why are we uppercasing these values?

Reply:
These values are extracted from the connection string, and they happen to be capitalized in there. I didn't think there was enough value in writing code that lowercased these values since this is a sample and I didn't want to take away from the main point of how to convert one connection string to another :-)